src/Microsoft.Health.Dicom.Core/Features/Validation/DateValidation.cs

```diff
@@ -15,13 +15,13 @@ internal class DateValidation : IElementValidation
 {
     private const string DateFormatDA = "yyyyMMdd";
 
-    public void Validate(DicomElement dicomElement, bool withLeniency = false)
+    public void Validate(DicomElement dicomElement, ValidationStyle validationStyle = ValidationStyle.Strict)
     {
         string value = dicomElement.GetFirstValueOrDefault<string>();
         string name = dicomElement.Tag.GetFriendlyName();
-        if (withLeniency)
+        if (validationStyle == ValidationStyle.Default)
```

Comment:
Should there be a base type for validators on `string` types that fetch the value of a tag and optionally trim it based on `ValidationStyle`?

Reply:
I think `CA1822` won't trigger if we do something like:

```csharp
internal abstract class StringElementValidation : IElementValidation
{
    public sealed void Validate(DicomElement dicomElement, ValidationLevel validationLevel = ValidationLevel.Strict)
    {
        EnsureArg.IsNotNull(dicomElement, nameof(dicomElement));

        string name = dicomElement.Tag.GetFriendlyName();
        string value = dicomElement.GetFirstValueOrDefault<string>();
        if (!string.IsNullOrEmpty(value) && validationLevel == ValidationLevel.Strict)
            value = value.TrimEnd('\0');

        ValidateStringElement(name, value, validationLevel);
    }

    protected abstract void ValidateStringElement(string name, string value, ValidationLevel validationLevel);
}
```